### PR TITLE
Implement factory methods

### LINKED ISSUE #462: Alternative constructors (e.g. `read` / `from_file`)
Many of our classes can be initialised in different ways. Most often, there should be an alternative constructor of the form:
```python
@classmethod
def from_file(cls, file_path):
    # Open the file, read parameters, etc.
    ...
    return cls(parameters)
```

I find this solution much more elegant than dealing with it within `__init__`. The name of the method could `from_file`, `read`, `read_from`, `from_fits`, or any other variant. I don't have a clear preference, but I would certainly stick to a single convention throughout the code.

Same goes for the argument: we should choose whether we specify files by their full path, or we separate path and filename. Here I think I'd go for the full path, but I'm happy to comply with whatever we decide.

### PR DESCRIPTION
**Pull request by PabloCorcho**
Fixes #462

### CODE REVIEW

File: src/pykoala/corrections/atmospheric_corrections.py [hunk 1]
```diff
@@ -50,26 +50,27 @@ class AtmosphericExtCorrection(CorrectionBase):
     def __init__(self,
                  extinction_correction=None,
                  extinction_correction_wave=None,
```
Comment (paranoya) [inline src/pykoala/corrections/atmospheric_corrections.py:52]: Should `None` be the default?

File: src/pykoala/corrections/atmospheric_corrections.py [hunk 2]
```diff
@@ -50,26 +50,27 @@ class AtmosphericExtCorrection(CorrectionBase):
...
+                 verbose=False, **kwargs):
         self.verbose = verbose
         self.corr_print("Initialising Atm ext. correction model.")
 
         # Initialise variables
         self.extinction_correction = extinction_correction
         self.extinction_correction_wave = extinction_correction_wave
-        self.extinction_file = extinction_file
 
         if self.extinction_correction is None:
```
Comment (paranoya) [inline src/pykoala/corrections/atmospheric_corrections.py:61]: Shouldn't this code disappear?

File: src/pykoala/cubing.py [hunk 3]
```diff
@@ -570,29 +570,32 @@ class Cube(SpectraContainer):
...
-            self.hdul_extensions_map = {"INTENSITY": "INTENSITY",
-                                        "VARIANCE": "VARIANCE"}
-        if self.hdul is not None:
-            print("[Cube] Initialising cube from input HDUL")
-            self.hdul = hdul
-        elif file_path is not None:
-            self.load_hdul(file_path)
-        self.get_wcs_from_header()
+        if hdul_extensions_map is not None:
+            self.hdul_extensions_map = hdul_extensions_map
```
Comment (paranoya) [inline src/pykoala/cubing.py:580]: I'm not sure whether `hdul_extensions_map` should be a class or an instance variable

File: src/pykoala/cubing.py [hunk 4]
```diff
@@ -570,29 +570,32 @@ class Cube(SpectraContainer):
...
+        return self._hdul
+
+    @hdul.setter
+    def hdul(self, hdul):
+        assert isinstance(hdul, fits.HDUList)
+        self._hdul = hdul
+
     @property
     def intensity(self):
         return self.hdul[self.hdul_extensions_map['INTENSITY']].data
```
Comment (paranoya) [inline src/pykoala/cubing.py:601]: This seems a good idea, somewhat different from the behaviour of `RSS`. Should we implement the `hdul_extensions_map` scheme there?

Alternatively, `intensity` could always be an array, and the `hdul_extensions_map` would only be relevant when [reading and] writing FITS files.

File: src/pykoala/cubing.py [hunk 5]
```diff
@@ -619,6 +622,30 @@ def rss_intensity(self):
     def rss_variance(self):
         return np.reshape(self.variance, (self.variance.shape[0], self.variance.shape[1]*self.variance.shape[2])).T
 
+    @classmethod
+    def from_fits(cls, path, hdul_extension_map=None, **kwargs):
```
Comment (paranoya) [inline src/pykoala/cubing.py:626]: I wonder whether we should again follow the same structure as in `RSS`, i.e. instrument-specific cubes (even if we only implement `KOALA_Cube` for the time being).

File: src/pykoala/cubing.py (hunk 3 above)
Comment (PabloCorcho) [reply src/pykoala/cubing.py:581]: a new class variable `default_hdul_extension_map`  is replacing the previous one.

File: src/pykoala/cubing.py (hunk 4 above)
Comment (PabloCorcho) [reply src/pykoala/cubing.py:606]: Indeed, we need to create an standard format for the RSS files processed with pykoala. Including the map sounds reasonable.

Let's see if this argument is convincing. Making `intensity` an array implies to allocate the memory space by default. Instead, if the data is read from the HDUL, only if when calling variable the memory is allocated...In any case, this is still a part of the architecture that is poorly defined, and I believe we will end up modifying it.

File: src/pykoala/cubing.py (hunk 5 above)
Comment (PabloCorcho) [reply src/pykoala/cubing.py:639]: It's doable, but I see that feature as a mid-term addition once the corrections are well polished.

**Pull request merged**